apps/files/utils.py

```diff
@@ -542,6 +542,10 @@ def parse_xpi(xpi, addon=None, check=True):
 def check_xpi_info(xpi_info, addon=None):
     from addons.models import Addon, BlacklistedGuid
     guid = xpi_info['guid']
+    if amo.get_user():
+        user_id = amo.get_user().id
+    else:
+        user_id = 0
```

Comment:
I'd rather have something like:
```python
if amo.get_user():
    user_addons = Addon.unfiltered.exclude(authors__id=amo.get_user()).filter(guid=guid)
else:
    user_addons = Addon.objects.none()
```
And then on line 562 use `user_addons.exists()` instead, what do you think?

I hope we can trust `amo.get_user()` in all the cases we're trying to create a new addon...